invites/async/reply.js

```diff
@@ -1,39 +1,75 @@
+const getContent = require('ssb-msg-content')
+const pull = require('pull-stream')
+const { heads } = require('ssb-sort')
+
 const {
+  isInvite,
   isReply,
   parseReply,
   versionStrings: {
     V1_SCHEMA_VERSION_STRING
   }
-} = require('scuttle-invite-schema')
-
+} = require('ssb-invite-schema')
+const buildError = require('../../lib/buildError')
 
 module.exports = function (server) {
   const getInvite = require('./getInvite')(server)
 
-  return function reply (params, callback) {
-    const reply = Object.assign({}, {
-      type: 'invite-reply',
-    }, params, {
-      version: V1_SCHEMA_VERSION_STRING
-    })
-
-    if (!reply.recps) reply.recps = []
-    if (!reply.recps.includes(server.id)) reply.recps = [...reply.recps, server.id]
+  return function reply (inviteKey, params, callback) {
+    getInvite(inviteKey, (err, invite) => {
+      if (err) return callback(err)
+      if (!isInvite(invite)) return callback(new Error(`${inviteKey} is not a valid invite, cannot reply to this`))
 
-    if (!isReply(reply)) {
-      var errors = reply.errors.map(e => e.field).join(', ')
-      return callback(new Error(`invalid: ${errors}`))
-    }
+      const { recps = [], root } = getContent(invite)
 
-    getInvite(reply.branch, (err, invite) => {
-      if (err) return callback(err)
-      const { value: { content: { recps } } } = invite
-      var whoami = server.whoami()
-      let recipients = recps.filter(recp => recp !== whoami.id)
-      var notInvited = recipients.length !== 1
-      if (notInvited) return callback(new Error(`invalid: you are not invited`))
+      const iWasInvited = Boolean(recps.find(recp => recp === server.id))
+      if (!iWasInvited) return callback(new Error(`invalid: you are not invited`))
+        
+      const reply = Object.assign({},
+        params,
+        { 
+          type: 'invite-reply',
+          version: V1_SCHEMA_VERSION_STRING,
+          recps,
+          root,
+          branch: inviteKey
+        }
+      )
+      if (!isReply(reply)) return callback(buildError(reply))
 
       server.publish(reply, callback)
+
+      // pull(
+      //   backlinksSource(root),
+      //   pull.collect((err, msgs) => {
+      //     if (err) return callback(err)
+
+      //     const reply = Object.assign({},
+      //       params,
+      //       { 
+      //         type: 'invite-reply',
+      //         version: V1_SCHEMA_VERSION_STRING,
+      //         recps,
+      //         root,
+      //         branch: heads(msgs)
```

Comment:
this works, but the ssb-invite-schema would need modifying to allow arrays of msgIds ... which is valid
This is most relevant to public invite / replies